.github/workflows/main.yml

```diff
@@ -12,41 +12,40 @@ on:
 
 jobs:
   build-and-test:
-    name: Test PY=${{ matrix.python-version }}, DEV=${{ matrix.developer }}, EXP=${{ matrix.experimental }}, DEP=${{ matrix.deprecated }}
-    runs-on: ubuntu-latest
+    name: Test PY=${{ matrix.python-version }}, BCD=${{ matrix.bitcoind-version }}, EXP=${{ matrix.experimental }}, DEP=${{ matrix.deprecated }}
+    runs-on: ubuntu-22.04
     timeout-minutes: 60
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.7, 3.8, 3.9]
-        developer: [0,1]
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
+        bitcoind-version: ["22.0", "26.0"]
         experimental: [1]
         deprecated: [0]
 
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
 
-    - name: Checkout c-lightning@master
-      uses: actions/checkout@v3
+    - name: Checkout c-lightning@v23.11
+      uses: actions/checkout@v4
       with:
         repository: 'ElementsProject/lightning'
         path: 'lightning'
-        ref: 'master'
+        ref: 'v23.11'
         submodules: 'recursive'
         fetch-depth: 0  # Required for pyln versions to be recognized
 
     - name: Download runtime dependencies
       run: |
-        export BITCOIND_VERSION="0.20.1"
-        wget https://storage.googleapis.com/c-lightning-tests/bitcoin-${BITCOIND_VERSION}-x86_64-linux-gnu.tar.bz2
-        tar -xjf bitcoin-${BITCOIND_VERSION}-x86_64-linux-gnu.tar.bz2
+        export BITCOIND_VERSION=${{ matrix.bitcoind-version }}
+        wget https://bitcoincore.org/bin/bitcoin-core-${BITCOIND_VERSION}/bitcoin-${BITCOIND_VERSION}-x86_64-linux-gnu.tar.gz
```

Comment:
Not sure why we used to use a Google storage bucket here and avoided using bitcoincore.org, but this is definitely the simplest fix for this.